src/miscwrappers.cpp

```diff
@@ -306,12 +308,14 @@ syscall(long sys_num, ...)
     break;
   }
 # ifndef __aarch64__
+# ifndef __riscv
```

Comment:
First of all, is SYS_open (__NR_open) is not defined, but __NR_openat is defined.
See 'man opeanat'.  So, let's add a FIXME for this.
Second, please use preprocessor commands with `!` and `||` instead of two commands.  It's more readable.

So, please change all of this to:
> `# if ! defined(__aarch64__) || ! defined(__riscv)`
> `// FIXME: For aarch64 and riscv, we should define this, below, as:`
> `//             ret = openat(AT_FDCWD,pathname, flags, mode);`

And finally, let';s change the two 'endif' statements to a single one:
> `# endif // if ! defined(__aarch64__) || ! defined(__riscv)`